lib/async_hooks.js

```diff
@@ -250,8 +254,12 @@ class AsyncResource {
         __proto__: null,
         configurable: true,
         enumerable: true,
-        value: this,
-        writable: true,
+        get: deprecate(function() {
+          return self;
+        }, 'The asyncResource property on bound functions is deprecated', 'DEP0172'),
+        set: deprecate(function(val) {
+          self = val;
+        }, 'The asyncResource property on bound functions is deprecated', 'DEP0172'),
```

Comment:
Probably a dummy question 😅 but I wonder why we didn't wrap the whole `bind` function with deprecate? (just for my curiosity)

Reply:
We're not deprecating the bind method. We're deprecating the asyncResource property it adds to the returned function